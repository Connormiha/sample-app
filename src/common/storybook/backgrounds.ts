import 'common/storybook/background.pcss';

export const backgrounds = {
    backgrounds: [
        {name: 'light-gray', value: '#f5f6fa', default: true},
        {name: 'White', value: '#fff'},
        {name: 'Black', value: '#000'},
        {name: 'Gray', value: '#ccc'},
        {name: 'Twitter', value: '#00aced'},
        {name: 'Facebook', value: '#3b5998'},
        {name: 'Ok.ru', value: '#ee7808'}
    ]
};
