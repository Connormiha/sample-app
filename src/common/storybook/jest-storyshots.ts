import initStoryshots from '@storybook/addon-storyshots';

export const requireSnapshot = (path: string): void => {
    const config: any = {
        config: ({configure}: any): any => {
            configure(() => {
                require(path.replace('.test', '.story'));
            }, module);
        },
        framework: 'react'
    };

    initStoryshots(config);
};
