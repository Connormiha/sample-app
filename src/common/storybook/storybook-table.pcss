.storybook-table {
    padding: 0 20px 20px;

    &__title {
        font-size: 15px;
    }

    &__table {
        border-collapse: collapse;
    }

    &__td {
        border: 0;
        padding: 0 20px 0 0;
        vertical-align: top;

        &_width {
            &_small {
                width: 100px;
            }

            &_medium {
                width: 170px;
            }

            &_large {
                width: 370px;
            }
        }
    }
}
