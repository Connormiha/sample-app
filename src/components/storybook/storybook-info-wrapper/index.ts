export * from './storybook-info-wrapper';
export {default} from './storybook-info-wrapper';
