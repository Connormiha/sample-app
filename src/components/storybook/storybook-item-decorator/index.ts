export * from './storybook-item-decorator';
export {default} from './storybook-item-decorator';
