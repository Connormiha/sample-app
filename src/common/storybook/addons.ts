import 'common/css/global.pcss';
import 'common/css/ui-variables.pcss';
import 'common/css/ui-animations.pcss';
import 'common/storybook/global.pcss';
import {addParameters, setAddon} from '@storybook/react';
import {backgrounds} from './backgrounds';
import JSXAddon from 'storybook-addon-jsx';

addParameters(backgrounds);
setAddon(JSXAddon);
