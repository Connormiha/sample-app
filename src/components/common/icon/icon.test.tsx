import {requireSnapshot} from 'common/storybook/jest-storyshots';

requireSnapshot(__filename);
