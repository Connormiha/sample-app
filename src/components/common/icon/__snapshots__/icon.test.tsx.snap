// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Storyshots Icon Large 1`] = `
Array [
  <span
    style={
      Object {
        "margin": "4px",
      }
    }
  >
    <span
      className="icon icon_name_copy icon_size_large"
    />
  </span>,
  <span
    style={
      Object {
        "margin": "4px",
      }
    }
  >
    <span
      className="icon icon_name_logo icon_size_large"
    />
  </span>,
]
`;

exports[`Storyshots Icon Medium 1`] = `
Array [
  <span
    style={
      Object {
        "margin": "4px",
      }
    }
  >
    <span
      className="icon icon_name_copy icon_size_medium"
    />
  </span>,
  <span
    style={
      Object {
        "margin": "4px",
      }
    }
  >
    <span
      className="icon icon_name_logo icon_size_medium"
    />
  </span>,
]
`;

exports[`Storyshots Icon Small 1`] = `
Array [
  <span
    style={
      Object {
        "margin": "4px",
      }
    }
  >
    <span
      className="icon icon_name_copy icon_size_small"
    />
  </span>,
  <span
    style={
      Object {
        "margin": "4px",
      }
    }
  >
    <span
      className="icon icon_name_logo icon_size_small"
    />
  </span>,
]
`;
